include/nbl/builtin/hlsl/scan/scheduler.hlsl

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_HLSL_SCAN_SCHEDULER_INCLUDED_
+#define _NBL_HLSL_SCAN_SCHEDULER_INCLUDED_
+
+#include "nbl/builtin/hlsl/scan/declarations.hlsl"
+#include "nbl/builtin/hlsl/scan/descriptors.hlsl"
+#include "nbl/builtin/hlsl/workgroup/basic.hlsl"
+#include "nbl/builtin/hlsl/workgroup/broadcast.hlsl"
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace scan
+{
+
+template<uint16_t WorkgroupSize>
+struct Scheduler
+{
+  
+  static Scheduler create(NBL_CONST_REF_ARG(Parameters_t) params, NBL_CONST_REF_ARG(DefaultSchedulerParameters_t) schedParams)
+  {
+      Scheduler<WorkgroupSize> scheduler;
+      scheduler.params = params;
+      scheduler.schedParams = schedParams;
+      return scheduler;
+  }
+    
+  template<class Accessor>
+  bool getWork(NBL_REF_ARG(Accessor) accessor)
```

Comment:
whats the accessor for? needs a better name

Also why don't you store the accessor past the `create` as a member?